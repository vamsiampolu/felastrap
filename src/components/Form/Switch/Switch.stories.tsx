import type { Meta, StoryObj, StoryFn } from "@storybook/react";
import { Switch } from "./Switch";
import { useState } from "react";

export type Story = StoryObj<typeof Switch>;

const meta: Meta<typeof Switch> = {
  title: "components/Form/Switch",
  component: Switch,
  parameters: {
    layout: "centered",
  },
};

export default meta;

export const Template: StoryFn<typeof Switch> = (args) => {
  const {
    id = "template-switch",
    label = "Switch",
    checked: initialChecked = false,
    ...props
  } = args;
  const [checked, setChecked] = useState(initialChecked);
  const onChange = () => {
    setChecked((currentChecked) => !currentChecked);
  };

  return (
    <Switch
      id={id}
      label={label}
      checked={checked}
      onChange={onChange}
      {...props}
    />
  );
};

export const Basic: Story = Template.bind({});
Basic.args = {
  label: "Switch Control",
  id: "basic-switch",
  checked: false,
};

export const Disabled: Story = Template.bind({});
Disabled.args = {
  label: "Disabled",
  id: "disabled-switch",
  checked: true,
  disabled: true,
};

export const Reverse: Story = Template.bind({});
Reverse.args = {
  label: "Reverse",
  id: "disabled-switch",
  checked: true,
  reverse: true,
};
