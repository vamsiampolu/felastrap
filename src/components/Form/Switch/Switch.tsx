import toggle from "/toggle.svg";
import toggleFocus from "/toggle-focus.svg";
import toggleChecked from "/toggle-checked.svg";
import { useFela } from "react-fela";
import { CheckboxLabel as Label } from "../Checkbox";
import { StyleProps } from "../../common";
import {
  ChangeEvent,
  ComponentPropsWithoutRef,
  ReactNode,
  useState,
} from "react";

const wrapperRule = ({ reverse }: StyleProps<{ reverse: boolean }>) => ({
  display: "block",
  minHeight: "1.5rem",
  paddingLeft: reverse ? 0 : "2.5em",
  paddingRight: reverse ? "2.5em" : 0,
  marginBottom: "0.125rem",
  textAlign: reverse ? "right" : "left",
});

const inputRule = ({ reverse }: StyleProps<{ reverse: boolean }>) => ({
  float: reverse ? "right" : "left",
  flexShrink: 0,
  backgroundImage: `url(${toggle})`,
  height: "1em",
  marginTop: "0.25em",
  verticalAlign: "top",
  appearance: "none",
  backgroundColor: "#fff",
  backgroundRepeat: "no-repeat",
  backgroundSize: "contain",
  border: "1px solid #dee2e6",
  colorAdjust: "exact",
  printColorAdjust: "exact",
  width: "2em",
  marginLeft: reverse ? 0 : "-2.5em !important",
  marginRight: reverse ? "-2.5em !important" : 0,
  backgroundPosition: "left center",
  borderRadius: "2em",
  transition: "background-position 0.15s ease-in-out",
  "@media (prefers-reduced-motion: reduce)": {
    transition: "none",
  },
  ":active": {
    filter: "brightness(90%)",
  },
  ":checked": {
    backgroundPosition: "right center",
    backgroundImage: `url(${toggleChecked})`,
    backgroundColor: "#0d6efd",
    borderColor: "#0d6efd",
  },
  ":disabled": {
    pointerEvents: "none",
    filter: "none",
    opacity: 0.5,
  },
  ":focus": {
    backgroundImage: `url(${toggleFocus})`,
    borderColor: "#86b7fe",
    boxShadow: "0 0 0 0.25rem rgba(13, 110, 253, 0.25)",
    outline: 0,
  },
});

type Attributes =
  | "id"
  | "name"
  | "className"
  | "checked"
  | "disabled"
  | "defaultChecked"
  | "onSelect"
  | "onChange"
  | "onInput"
  | "required";

export type SwitchProps = Pick<
  ComponentPropsWithoutRef<"input">,
  Attributes
> & { reverse?: boolean; label: ReactNode };

export const Switch = ({
  id,
  className = "",
  disabled = false,
  reverse = false,
  required = false,
  checked: controlledChecked,
  name,
  onSelect,
  onChange,
  onInput,
  label,
  defaultChecked,
  ...props
}: SwitchProps) => {
  const { css } = useFela({ reverse });

  const [uncontrolledChecked, setUncontrolledChecked] = useState(
    defaultChecked ?? false,
  );
  const isControlled = controlledChecked !== undefined;
  const checked = isControlled ? controlledChecked : uncontrolledChecked;

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    if (!isControlled) {
      const updatedValue = e.target.checked;
      setUncontrolledChecked(updatedValue);
    }

    if (onChange) {
      onChange(e);
    }
  };

  const externalClassNames = "form-check-input form-switch-input";
  const externalWrapperClassNames = "form-check form-switch";

  const wrapperClassName = `${css(
    wrapperRule,
  )} ${externalWrapperClassNames}`.trim();
  const inputClassName = `${css(
    inputRule,
  )} ${externalClassNames} ${className}`.trim();

  return (
    <div data-testid="switch-field" className={wrapperClassName}>
      <input
        id={id}
        data-testid="switch-input"
        className={inputClassName}
        checked={checked}
        disabled={disabled}
        type="checkbox"
        required={required}
        name={name}
        onSelect={onSelect}
        onChange={handleChange}
        onInput={onInput}
        role="switch"
        {...props}
      />
      <Label className="form-check-label form-switch-label" htmlFor={id!}>
        {label}
      </Label>
    </div>
  );
};
