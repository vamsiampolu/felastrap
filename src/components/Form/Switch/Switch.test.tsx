import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { IRenderer } from "fela";
import { ChangeEvent, useState } from "react";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../test-helpers";
import { Switch, SwitchProps } from "./Switch";

function ControlledSwitch({
  checked: initialChecked,
  onChange,
  label,
  ...props
}: SwitchProps) {
  const [checked, setChecked] = useState(initialChecked ?? false);

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setChecked(e.target.checked);
    if (onChange) {
      onChange(e);
    }
  };

  return (
    <Switch
      checked={checked}
      onChange={handleChange}
      label={label}
      {...props}
    />
  );
}

describe("components/Form/Switch", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should display a switch button", async () => {
    const component = <Switch id="spiel" label="Switch Label" />;

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("switch-field")).toBeInTheDocument();

    expect(screen.getByTestId("switch-input")).toBeInTheDocument();
    expect(screen.getByTestId("form-label")).toBeInTheDocument();

    expect(screen.queryByTestId("switch-input")).toHaveAttribute("id", "spiel");
    expect(screen.queryByTestId("form-label")).toHaveAttribute("for", "spiel");

    expect(screen.queryByTestId("switch-input")).toHaveAttribute(
      "role",
      "switch",
    );
    expect(screen.queryByTestId("switch-input")).toHaveAttribute(
      "type",
      "checkbox",
    );

    expect(screen.queryByTestId("switch-input")?.tagName).toBe("INPUT");
    expect(screen.queryByTestId("switch-input")).not.toBeChecked();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should reverse a switch button", async () => {
    const component = <Switch reverse id="spiel" label="Switch Label" />;

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should display a disabled switch button", async () => {
    const component = <Switch disabled id="spiel" label="Switch Label" />;
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("switch-input")).toBeInTheDocument();
    expect(screen.queryByTestId("switch-input")).toBeDisabled();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should display a checked switch button", async () => {
    const component = (
      <Switch
        defaultChecked
        onChange={() => undefined}
        id="spiel"
        label="Switch Label"
      />
    );
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("switch-input")).toBeInTheDocument();
    expect(screen.queryByTestId("switch-input")).toBeChecked();

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should change the switch state to checked", async () => {
    const onChange = jest.fn();
    const component = (
      <Switch onChange={onChange} id="spiel" label="Switch Label" />
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    const user = userEvent.setup();

    expect(screen.getByTestId("switch-input")).toBeInTheDocument();
    expect(screen.queryByTestId("switch-input")).not.toBeChecked();

    const checkbox = screen.queryByTestId("switch-input");
    await user.click(checkbox!);

    expect(checkbox).toBeChecked();
    expect(onChange).toHaveBeenCalled();
  });

  it("should change the switch state to unchecked", async () => {
    const onChange = jest.fn();
    const component = (
      <Switch
        defaultChecked
        onChange={onChange}
        id="spiel"
        label="Switch Label"
      />
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    const user = userEvent.setup();

    expect(screen.getByTestId("switch-input")).toBeInTheDocument();
    expect(screen.queryByTestId("switch-input")).toBeChecked();

    const checkbox = screen.queryByTestId("switch-input");
    await user.click(checkbox!);

    expect(screen.queryByTestId("switch-input")).not.toBeChecked();
    expect(onChange).toHaveBeenCalled();
  });

  it("should check a controlled switch", async () => {
    const onChange = jest.fn();
    const component = (
      <ControlledSwitch
        checked={false}
        onChange={onChange}
        id="spiel"
        label="Switch Label"
      />
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    const user = userEvent.setup();

    const checkbox = screen.queryByTestId("switch-input");
    await user.click(checkbox!);

    expect(screen.queryByTestId("switch-input")).toBeChecked();
    expect(onChange).toHaveBeenCalled();
  });

  it("should uncheck a controlled switch", async () => {
    const onChange = jest.fn();
    const component = (
      <ControlledSwitch
        checked
        onChange={onChange}
        id="spiel"
        label="Switch Label"
      />
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    const user = userEvent.setup();

    const checkbox = screen.queryByTestId("switch-input");
    await user.click(checkbox!);

    expect(checkbox).not.toBeChecked();
    expect(onChange).toHaveBeenCalled();
  });
});
