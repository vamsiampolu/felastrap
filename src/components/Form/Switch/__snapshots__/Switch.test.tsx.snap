// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form/Switch should display a checked switch button 1`] = `
".a {
  display: block;
}
.b {
  min-height: 1.5rem;
}
.c {
  padding-left: 2.5em;
}
.d {
  padding-right: 0;
}
.e {
  margin-bottom: 0.125rem;
}
.f {
  text-align: left;
}
.g {
  float: left;
}
.h {
  flex-shrink: 0;
}
.i {
  background-image: url(test-file-stub);
}
.j {
  height: 1em;
}
.k {
  margin-top: 0.25em;
}
.l {
  vertical-align: top;
}
.m {
  appearance: none;
}
.n {
  background-color: #fff;
}
.o {
  background-repeat: no-repeat;
}
.p {
  background-size: contain;
}
.q {
  color-adjust: exact;
}
.r {
  print-color-adjust: exact;
}
.s {
  width: 2em;
}
.t {
  margin-left: -2.5em !important;
}
.u {
  margin-right: 0;
}
.v {
  background-position: left center;
}
.ac:checked {
  background-position: right center;
}
.ae:checked {
  background-image: url(test-file-stub);
}
.af:checked {
  background-color: #0d6efd;
}
.ag:checked {
  border-top-color: #0d6efd;
}
.ah:checked {
  border-right-color: #0d6efd;
}
.ai:checked {
  border-bottom-color: #0d6efd;
}
.aj:checked {
  border-left-color: #0d6efd;
}
.ak:disabled {
  pointer-events: none;
}
.al:disabled {
  filter: none;
}
.am:disabled {
  opacity: 0.5;
}
.au {
  transition-property: background-position;
}
.av {
  transition-duration: 0.15s;
}
.aw {
  transition-delay: ease-in-out;
}
.ax {
  transition-timing-function: ease;
}
.ay {
  border-top-left-radius: 2em;
}
.az {
  border-top-right-radius: 2em;
}
.ba {
  border-bottom-right-radius: 2em;
}
.bb {
  border-bottom-left-radius: 2em;
}
.bc {
  border-top-width: 1px;
}
.bd {
  border-right-width: 1px;
}
.be {
  border-bottom-width: 1px;
}
.bf {
  border-left-width: 1px;
}
.bg {
  border-top-style: solid;
}
.bh {
  border-right-style: solid;
}
.bi {
  border-bottom-style: solid;
}
.bj {
  border-left-style: solid;
}
.bk {
  border-top-color: #dee2e6;
}
.bl {
  border-right-color: #dee2e6;
}
.bm {
  border-bottom-color: #dee2e6;
}
.bn {
  border-left-color: #dee2e6;
}
.bo {
  display: inline-block;
}
.bp {
  margin-bottom: 0.5rem;
}
.an:focus {
  background-image: url(test-file-stub);
}
.ao:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.ap:focus {
  outline-width: 0;
}
.aq:focus {
  border-top-color: #86b7fe;
}
.ar:focus {
  border-right-color: #86b7fe;
}
.as:focus {
  border-bottom-color: #86b7fe;
}
.at:focus {
  border-left-color: #86b7fe;
}
.ab:active {
  filter: brightness(90%);
}
@media (prefers-reduced-motion: reduce) {
  .w {
    transition-property: none;
  }
  .x {
    transition-duration: 0s;
  }
  .y {
    transition-delay: 0s;
  }
  .z {
    transition-timing-function: ease;
  }
}


<div data-testid="switch-field" class="a b c d e f form-check form-switch">
  <input
    id="spiel"
    data-testid="switch-input"
    class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn g h i j k l m n o p q r s t u v w x y z form-check-input form-switch-input"
    type="checkbox"
    role="switch"
    checked=""
  /><label
    data-testid="form-label"
    for="spiel"
    class="bo bp form-label form-check-label form-check-label form-switch-label"
    >Switch Label</label
  >
</div>
;
"
`;

exports[`components/Form/Switch should display a disabled switch button 1`] = `
".a {
  display: block;
}
.b {
  min-height: 1.5rem;
}
.c {
  padding-left: 2.5em;
}
.d {
  padding-right: 0;
}
.e {
  margin-bottom: 0.125rem;
}
.f {
  text-align: left;
}
.g {
  float: left;
}
.h {
  flex-shrink: 0;
}
.i {
  background-image: url(test-file-stub);
}
.j {
  height: 1em;
}
.k {
  margin-top: 0.25em;
}
.l {
  vertical-align: top;
}
.m {
  appearance: none;
}
.n {
  background-color: #fff;
}
.o {
  background-repeat: no-repeat;
}
.p {
  background-size: contain;
}
.q {
  color-adjust: exact;
}
.r {
  print-color-adjust: exact;
}
.s {
  width: 2em;
}
.t {
  margin-left: -2.5em !important;
}
.u {
  margin-right: 0;
}
.v {
  background-position: left center;
}
.ac:checked {
  background-position: right center;
}
.ae:checked {
  background-image: url(test-file-stub);
}
.af:checked {
  background-color: #0d6efd;
}
.ag:checked {
  border-top-color: #0d6efd;
}
.ah:checked {
  border-right-color: #0d6efd;
}
.ai:checked {
  border-bottom-color: #0d6efd;
}
.aj:checked {
  border-left-color: #0d6efd;
}
.ak:disabled {
  pointer-events: none;
}
.al:disabled {
  filter: none;
}
.am:disabled {
  opacity: 0.5;
}
.au {
  transition-property: background-position;
}
.av {
  transition-duration: 0.15s;
}
.aw {
  transition-delay: ease-in-out;
}
.ax {
  transition-timing-function: ease;
}
.ay {
  border-top-left-radius: 2em;
}
.az {
  border-top-right-radius: 2em;
}
.ba {
  border-bottom-right-radius: 2em;
}
.bb {
  border-bottom-left-radius: 2em;
}
.bc {
  border-top-width: 1px;
}
.bd {
  border-right-width: 1px;
}
.be {
  border-bottom-width: 1px;
}
.bf {
  border-left-width: 1px;
}
.bg {
  border-top-style: solid;
}
.bh {
  border-right-style: solid;
}
.bi {
  border-bottom-style: solid;
}
.bj {
  border-left-style: solid;
}
.bk {
  border-top-color: #dee2e6;
}
.bl {
  border-right-color: #dee2e6;
}
.bm {
  border-bottom-color: #dee2e6;
}
.bn {
  border-left-color: #dee2e6;
}
.bo {
  display: inline-block;
}
.bp {
  margin-bottom: 0.5rem;
}
.an:focus {
  background-image: url(test-file-stub);
}
.ao:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.ap:focus {
  outline-width: 0;
}
.aq:focus {
  border-top-color: #86b7fe;
}
.ar:focus {
  border-right-color: #86b7fe;
}
.as:focus {
  border-bottom-color: #86b7fe;
}
.at:focus {
  border-left-color: #86b7fe;
}
.ab:active {
  filter: brightness(90%);
}
@media (prefers-reduced-motion: reduce) {
  .w {
    transition-property: none;
  }
  .x {
    transition-duration: 0s;
  }
  .y {
    transition-delay: 0s;
  }
  .z {
    transition-timing-function: ease;
  }
}


<div data-testid="switch-field" class="a b c d e f form-check form-switch">
  <input
    id="spiel"
    data-testid="switch-input"
    class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn g h i j k l m n o p q r s t u v w x y z form-check-input form-switch-input"
    disabled=""
    type="checkbox"
    role="switch"
  /><label
    data-testid="form-label"
    for="spiel"
    class="bo bp form-label form-check-label form-check-label form-switch-label"
    >Switch Label</label
  >
</div>
;
"
`;

exports[`components/Form/Switch should display a switch button 1`] = `
".a {
  display: block;
}
.b {
  min-height: 1.5rem;
}
.c {
  padding-left: 2.5em;
}
.d {
  padding-right: 0;
}
.e {
  margin-bottom: 0.125rem;
}
.f {
  text-align: left;
}
.g {
  float: left;
}
.h {
  flex-shrink: 0;
}
.i {
  background-image: url(test-file-stub);
}
.j {
  height: 1em;
}
.k {
  margin-top: 0.25em;
}
.l {
  vertical-align: top;
}
.m {
  appearance: none;
}
.n {
  background-color: #fff;
}
.o {
  background-repeat: no-repeat;
}
.p {
  background-size: contain;
}
.q {
  color-adjust: exact;
}
.r {
  print-color-adjust: exact;
}
.s {
  width: 2em;
}
.t {
  margin-left: -2.5em !important;
}
.u {
  margin-right: 0;
}
.v {
  background-position: left center;
}
.ac:checked {
  background-position: right center;
}
.ae:checked {
  background-image: url(test-file-stub);
}
.af:checked {
  background-color: #0d6efd;
}
.ag:checked {
  border-top-color: #0d6efd;
}
.ah:checked {
  border-right-color: #0d6efd;
}
.ai:checked {
  border-bottom-color: #0d6efd;
}
.aj:checked {
  border-left-color: #0d6efd;
}
.ak:disabled {
  pointer-events: none;
}
.al:disabled {
  filter: none;
}
.am:disabled {
  opacity: 0.5;
}
.au {
  transition-property: background-position;
}
.av {
  transition-duration: 0.15s;
}
.aw {
  transition-delay: ease-in-out;
}
.ax {
  transition-timing-function: ease;
}
.ay {
  border-top-left-radius: 2em;
}
.az {
  border-top-right-radius: 2em;
}
.ba {
  border-bottom-right-radius: 2em;
}
.bb {
  border-bottom-left-radius: 2em;
}
.bc {
  border-top-width: 1px;
}
.bd {
  border-right-width: 1px;
}
.be {
  border-bottom-width: 1px;
}
.bf {
  border-left-width: 1px;
}
.bg {
  border-top-style: solid;
}
.bh {
  border-right-style: solid;
}
.bi {
  border-bottom-style: solid;
}
.bj {
  border-left-style: solid;
}
.bk {
  border-top-color: #dee2e6;
}
.bl {
  border-right-color: #dee2e6;
}
.bm {
  border-bottom-color: #dee2e6;
}
.bn {
  border-left-color: #dee2e6;
}
.bo {
  display: inline-block;
}
.bp {
  margin-bottom: 0.5rem;
}
.an:focus {
  background-image: url(test-file-stub);
}
.ao:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.ap:focus {
  outline-width: 0;
}
.aq:focus {
  border-top-color: #86b7fe;
}
.ar:focus {
  border-right-color: #86b7fe;
}
.as:focus {
  border-bottom-color: #86b7fe;
}
.at:focus {
  border-left-color: #86b7fe;
}
.ab:active {
  filter: brightness(90%);
}
@media (prefers-reduced-motion: reduce) {
  .w {
    transition-property: none;
  }
  .x {
    transition-duration: 0s;
  }
  .y {
    transition-delay: 0s;
  }
  .z {
    transition-timing-function: ease;
  }
}


<div data-testid="switch-field" class="a b c d e f form-check form-switch">
  <input
    id="spiel"
    data-testid="switch-input"
    class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn g h i j k l m n o p q r s t u v w x y z form-check-input form-switch-input"
    type="checkbox"
    role="switch"
  /><label
    data-testid="form-label"
    for="spiel"
    class="bo bp form-label form-check-label form-check-label form-switch-label"
    >Switch Label</label
  >
</div>
;
"
`;

exports[`components/Form/Switch should reverse a switch button 1`] = `
".a {
  display: block;
}
.b {
  min-height: 1.5rem;
}
.c {
  padding-left: 0;
}
.d {
  padding-right: 2.5em;
}
.e {
  margin-bottom: 0.125rem;
}
.f {
  text-align: right;
}
.g {
  float: right;
}
.h {
  -webkit-flex-shrink: 0;
  -ms-flex-shrink: 0;
  flex-shrink: 0;
}
.i {
  -webkit-background-image: url(test-file-stub);
  background-image: url(test-file-stub);
}
.j {
  height: 1em;
}
.k {
  margin-top: 0.25em;
}
.l {
  vertical-align: top;
}
.m {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}
.n {
  -webkit-background-color: #fff;
  background-color: #fff;
}
.o {
  background-repeat: no-repeat;
}
.p {
  background-size: contain;
}
.q {
  color-adjust: exact;
}
.r {
  print-color-adjust: exact;
}
.s {
  width: 2em;
}
.t {
  margin-left: 0;
}
.u {
  margin-right: -2.5em !important;
}
.v {
  background-position: left center;
}
.ac:checked {
  background-position: right center;
}
.ae:checked {
  -webkit-background-image: url(test-file-stub);
  background-image: url(test-file-stub);
}
.af:checked {
  -webkit-background-color: #0d6efd;
  background-color: #0d6efd;
}
.ag:checked {
  border-top-color: #0d6efd;
}
.ah:checked {
  border-right-color: #0d6efd;
}
.ai:checked {
  border-bottom-color: #0d6efd;
}
.aj:checked {
  border-left-color: #0d6efd;
}
.ak:disabled {
  pointer-events: none;
}
.al:disabled {
  -webkit-filter: none;
  filter: none;
}
.am:disabled {
  opacity: 0.5;
}
.au {
  transition-property: background-position;
}
.av {
  transition-duration: 0.15s;
}
.aw {
  transition-delay: ease-in-out;
}
.ax {
  transition-timing-function: ease;
}
.ay {
  border-top-left-radius: 2em;
}
.az {
  border-top-right-radius: 2em;
}
.ba {
  border-bottom-right-radius: 2em;
}
.bb {
  border-bottom-left-radius: 2em;
}
.bc {
  border-top-width: 1px;
}
.bd {
  border-right-width: 1px;
}
.be {
  border-bottom-width: 1px;
}
.bf {
  border-left-width: 1px;
}
.bg {
  border-top-style: solid;
}
.bh {
  border-right-style: solid;
}
.bi {
  border-bottom-style: solid;
}
.bj {
  border-left-style: solid;
}
.bk {
  border-top-color: #dee2e6;
}
.bl {
  border-right-color: #dee2e6;
}
.bm {
  border-bottom-color: #dee2e6;
}
.bn {
  border-left-color: #dee2e6;
}
.bo {
  display: inline-block;
}
.bp {
  margin-bottom: 0.5rem;
}
.an:focus {
  -webkit-background-image: url(test-file-stub);
  background-image: url(test-file-stub);
}
.ao:focus {
  box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
}
.ap:focus {
  outline-width: 0;
}
.aq:focus {
  border-top-color: #86b7fe;
}
.ar:focus {
  border-right-color: #86b7fe;
}
.as:focus {
  border-bottom-color: #86b7fe;
}
.at:focus {
  border-left-color: #86b7fe;
}
.ab:active {
  -webkit-filter: brightness(90%);
  filter: brightness(90%);
}
@media (prefers-reduced-motion: reduce) {
  .w {
    transition-property: none;
  }
  .x {
    transition-duration: 0s;
  }
  .y {
    transition-delay: 0s;
  }
  .z {
    transition-timing-function: ease;
  }
}


<div data-testid="switch-field" class="a b c d e f form-check form-switch">
  <input
    id="spiel"
    data-testid="switch-input"
    class="ab ac ae af ag ah ai aj ak al am an ao ap aq ar as at au av aw ax ay az ba bb bc bd be bf bg bh bi bj bk bl bm bn g h i j k l m n o p q r s t u v w x y z form-check-input form-switch-input"
    type="checkbox"
    role="switch"
  /><label
    data-testid="form-label"
    for="spiel"
    class="bo bp form-label form-check-label form-check-label form-switch-label"
    >Switch Label</label
  >
</div>
;
"
`;
